### ISSUE TITLE
[PAY 8/20] You can invite someone who's already a member of the workspace

### ISSUE THREAD

**Issue by aman-atg** [Expensify/App#4307]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a workspace
2. Choose people tabs
3. Invite yourself by adding your email address
4. It'll show two entries 

## Expected Result:
An error message
![image](https://user-images.githubusercontent.com/28243942/127530258-b9bf3b11-0c8d-4d04-a102-82a5e02f4f2a.png)

## Actual Result:
![image](https://user-images.githubusercontent.com/28243942/127528952-707f30b6-4f33-4497-889f-a175e390d737.png)

## Workaround:
Refreshing the browser make the 2nd entry disappear after a few seconds.

## Platform:
Where is this issue occurring?

- [x] Web
- [x] iOS
- [x] Android
- [x] Desktop App
- [x] Mobile Web

**Version Number:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @strepanier03 (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by aman-atg** [comment]
## Proposal 

https://github.com/Expensify/App/blob/9f15c144bd0c0c716ee4655f8e7946e30588ac1a/src/pages/workspace/WorkspaceInvitePage.js#L74-L85


* Adding the below code after line 81 will fix the issue
```js
    const isUserPresent = _.contains(logins, this.props.session.email);
        if (isUserPresent) {
            Growl.error(this.props.translate('workspace.invite.cantInviteYourself'), 5000);
            return;
        }
```

* `workspace.invite.cantInviteYourself` can be a message like `You can't invite yourself.` or something similar

**Comment by strepanier03** [comment]
Unique — Looks to be unique to me,
Issue hygiene — Looks go here as well,
Clarity of problem — Problem is stated clearly,
Value — Looks like something we should fix,

Passing to Eng.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @francoisl (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by francoisl** [comment]
Hi @aman-atg, your proposal would work for the very specific issue where someone can invite themselves, but the issue can be generalized and rephrased as: you can invite someone who's already a member of the workspace (e.g. invite the same account twice in a row).

With that in mind, can you please update your proposal? For the error message, I suggest something like `<login> is already a member of this workspace`.
Thanks!

**Comment by aman-atg** [comment]
@francoisl  Thanks for the input. I will update the Proposal with better logic that will cover the above mentioned scenario. 
 
I have one question, though. Since we're supporting multiple logins.. what should be the error message when 2 or more logins are duplicates.

**Comment by francoisl** [comment]
I agree that following the same convention and returning early would make sense. In this case, we could use `${login} is already a member of this workspace.` for the error message.

Although thinking about it more, I'm now thinking that in terms of providing a better UX, maybe we could just silently ignore and filter out the logins that are already in the workspace (and if all the logins are already in the workspace, we don't call `invite()` at all)? This way, the user doesn't see an error message, and they still "accomplished" what they were trying to do.

**Comment by aman-atg** [comment]
> Although thinking about it more, I'm now thinking that in terms of providing a better UX, maybe we could just silently ignore and filter out the logins that are already in the workspace

* IMO that's a valid option for us.

> (and if all the logins are already in the workspace, we don't call `invite()` at all)? This way, the user doesn't see an error message, and they still "accomplished" what they were trying to do.

* Here, however I feel that if all the `logins` entered by the `User` are `duplicate` then `User` should get some feedback like 
```js
     // `Please ensure that the user is not already a member of the workspace` or
     // `Please ensure that the user is not already invited.`
```

**Comment by francoisl** [comment]
I'm not passionate about either solution, I'm going to relabel this to get another opinion.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by mananjadhav** [comment]
@francoisl I feel the input itself is not user-friendly.  There is no way to know that you can invite more than one person by comma separating values.


**Proposal**
<img src="https://user-images.githubusercontent.com/3069065/127727605-04d3b175-a8d8-4b63-a3d7-c8a26268c232.png" height="200" />

My approach would be to change the input to a tagged input such as above. I can see Workspace is in beta so there's a lot we can do to improve the UX here.

This will handle the cases for multiple inputs as well as filtering the already existing participants.

btw the current flow should also check if the phone number entered is of the currently logged-in user.

**Comment by aman-atg** [comment]
> This will handle the cases for multiple inputs

@mananjadhav this is being dealt with on a separate issue #4029

**Comment by MelvinBot** [comment]
Triggered auto assignment to @deetergp (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by laurenreidexpensify** [comment]
@deetergp can you take a second look at this one and confirm if I can upload to Upwork? Thanks

**Comment by deetergp** [comment]
Apologies, I've a bit bogged down, looking at it now…

**Comment by deetergp** [comment]
The code portion of the proposal looks solid, but I'm not sold on any of the error message. How about something like `That user is already a member of this workspace`.

**Comment by aman-atg** [comment]
> As far as the message goes, how about something like `That user is already a member of this workspace`?

Sounds good to me!

**Comment by laurenreidexpensify** [comment]
Nice so @deetergp to confirm - this can move to upwork and we are going with @aman-atg's proposal to fix it?

**Comment by deetergp** [comment]
@laurenreidexpensify Yes, let's do it!

**Comment by deetergp** [comment]
Waiting on a PR. Do you think this should stay a Daily, @laurenreidexpensify, or is setting to Weekly more appropriate?

**Comment by aman-atg** [comment]
Added a PR. Not yet hired on upwork though.

**Comment by deetergp** [comment]
Setting this back to `Planning`

**Comment by aman-atg** [comment]
Accepted. ([Proposal](https://www.upwork.com/ab/proposals/1425733918605635584))

**Comment by deetergp** [comment]
The job has been offered & accepted and the PR has been merged. I think we're good here 👍

**Comment by mallenexpensify** [comment]
_Officially_ hired on Upwork. Will pay on 8/18 if no regressions. Bumped to Weekly

**Comment by mallenexpensify** [comment]
Paid in Upwork with bonus for reporting issues. Deployed to production 3 days ago, closing
